docs/source/developers/python.rst

```diff
@@ -509,7 +509,7 @@ First, starting from a fresh clone of Apache Arrow:
          --file arrow\ci\conda_env_cpp.txt ^
```

Comment:
Here, we don't include `conda_env_unix.txt` since that are Linux/MacOS specific packages. 
But, I see that that file includes `ccache` and there are windows conda packages for this. Should we move that package to the `conda_env_cpp.txt`? (or otherwise list here manually)

Reply:
I would add proposed changes in the scope of https://issues.apache.org/jira/browse/ARROW-17949

- The C++ documentation needs to be updated https://arrow.apache.org/docs/dev/developers/cpp/windows.html#building-with-ninja-and-clcache
- Python dev page can be updated with adding `ccache` (manually or to `conda_env_cpp.txt`), some CMake flags would also be needed (`ARROW_USE_CCACHE` and maybe [`ARROW_USE_SCCACHE`](https://github.com/apache/arrow/blob/89c0214fa43f8d1bf2e19e3bae0fc3009df51e15/cpp/cmake_modules/DefineOptions.cmake#L174-L175)?) plus I would test it locally so that nothing is missing.

What do you think?